### PR TITLE
Adsk Contrib - Add grading ops

### PR DESCRIPTION
**Pull request by BernardLefebvre**
The adds the grading transforms described in issue #1100 

- GradingPrimary
- GradingRGBCurve
- GradingTone

Inverse transform for GradingRGBCurve and GradingTone still have to be implemented.

Note that following the requirement introduced by the new grading transforms, GLSL fragment program generated by OCIO has now to support the static array intialization which requires the GLSL v1.2 at minimum (it means OpenGL 2.1 published in 2006). So, the minimal GLSL version is now 1.2. As side-effect any generated fragment shader program consuming OCIO generated code must now add the version directive with the appropriate version to avoid being in GLSL 1.1 (which is the behavior when no version directive is present).
 
Signed-off-by: Bernard Lefebvre <bernard.lefebvre@autodesk.com>

### CODE REVIEW

**Comment by hodoulp** [comment]
Found in the [GLSL version 1.20](https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.1.20.pdf) that `uniform` can now be statically initialized: `Initializers may only be used in declarations of globals with no storage qualifier, a const qualifier, or a uniform qualifier` from section 4.3.
That could be used in OCIO when generating the fragment shader program which then simplifies the OpenGL helper library.

File: share/cmake/modules/Findyaml-cpp.cmake [hunk 1]
```diff
@@ -29,8 +29,24 @@ if(NOT OCIO_INSTALL_EXT_PACKAGES STREQUAL ALL)
     set(_yaml-cpp_REQUIRED_VARS yaml-cpp_LIBRARY)
```
Comment (michdolan) [inline share/cmake/modules/Findyaml-cpp.cmake:29]: This change seems like it slipped in from Patrick's CMake work which we resolved. Should likely revert this file.

File: src/bindings/python/PyGradingData.cpp [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+    for (size_t pt = 0; pt < numPt; ++pt)
+    {
+        to->getControlPoint(pt) = from->getControlPoint(pt);
+    }
+}
+}
+
+void bindPyGradingData(py::module & m)
+{
+    py::class_<GradingRGBM>(m, "GradingRGBM")
```
Comment (michdolan) [inline src/bindings/python/PyGradingData.cpp:33]: TODO: Add constructor with keyword arg support to all Grading classes (can be done in later PR).

File: share/cmake/modules/Findyaml-cpp.cmake (hunk 1 above)
Comment (BernardLefebvre) [reply share/cmake/modules/Findyaml-cpp.cmake:29]: Will do.

File: src/bindings/python/PyGradingData.cpp (hunk 2 above)
Comment (BernardLefebvre) [reply src/bindings/python/PyGradingData.cpp:33]: Ok

File: share/cmake/modules/Findyaml-cpp.cmake (hunk 1 above)
Comment (michdolan) [reply share/cmake/modules/Findyaml-cpp.cmake]: There's still changes being made to this file in the most recent commit. I think the state the file is in master currently is working well so would be good to leave it unchanged.

Comment (BernardLefebvre) [reply share/cmake/modules/Findyaml-cpp.cmake]: I had tried to revert my original local commit but that might have interfere with the merge from master. Next commit will replace these files with their versions in master.

**Comment by hodoulp** [comment]
Following the two-weeks rule, here is a friendly reminder that the branch can now be merged at the end of this week unless there are new comments.

**Comment by KelSolaar** [comment]
Hello,

I wanted to report that following my [comment in issue #1100](https://github.com/AcademySoftwareFoundation/OpenColorIO/issues/1100#issuecomment-674340278), a few people, that shall remain nameless for now, from the industry, stakeholders in their respective studios/companies, have reached to me and raised concern about the lack of transparency in the process that ending up having the 22K lines of code of that PR landing like a _fly in the ointment_ if I can say so.

The fact that the two-weeks rule is being invoked here makes it even more worry-some. Again, I would like to make an emphasis on the fact that I'm strictly talking about the process used to deliver the feature and not the content of the feature itself.

Cheers,

Thomas

PS: I will create a separate issue to centralise the discussions pertaining to the problem being discussed here.

**Comment by doug-walker** [comment]
Hi @KelSolaar ,

As always, we will not merge if there are ongoing discussions.  It was simply not clear to us that there were still topics left to discuss about the PR itself.

Thank you for creating issue #1124, so that we can discuss the development process.  

If someone would like to actually discuss the new transforms for building looks, please make a comment (either here or in issue #1100).

Doug

**Comment by KelSolaar** [comment]
Hi @doug-walker,

Your explanations about the scope in #1100 made sense to me and I'm certainly not opposed to merging the PR but it would have been disingenuous from my part to not mention the _private_ discussions that occurred lately.

> Thank you for creating issue #1124, so that we can discuss the development process.

You are very much welcome! I should have done it earlier but... time... you know :)

**Comment by michdolan** [comment]
@KelSolaar and others. I know there are some reservations about this work in terms of OCIO's scope. Are there objections to merging this and working to clearly communicate the intent of these transforms in documentation?

**Comment by KelSolaar** [comment]
@michdolan : As mentioned above, I'm fine. I'm still a bit concerned by the fact we have opened a Pandora Box here but fine nonetheless :)

**Review by doug-walker — APPROVED** [review]

**Comment by doug-walker** [comment]
Thank you Michael for your review and Thomas for the interesting discussions.  Given that the PR has been available for review/comment for over three weeks and there are no ongoing active discussions, we will proceed with the merge.

**Pull request merged**